### PR TITLE
Product page funtionality

### PR DESCRIPTION
**Pull request by bhtibrewal**
did API calls on product page

### CODE REVIEW

File: src/context/ProductContext.jsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  // const fetchProduct = async () => {
+  //   try {
+  //     const res = await axios.get("api/products");
+  //     if (res.status === 200) {
+  //       setProductList(res.data.products);
+  //     }
+  //   } catch {
+  //     throw new Error("error");
+  //   }
+  // };
```
Comment (Naman-Saxena1) [inline src/context/ProductContext.jsx:22]: Can remove commented code if not planning to use it in the future.

File: src/pages/home_page/home_page.css [hunk 2]
```diff
@@ -16,6 +15,16 @@
...
+.hero-content {
+  padding: 4rem;
+  background: rgba(0, 0, 0, 0.36);
+  display: grid;
+  place-items: center;
+}
+.hero-content > .btn {
+  font-size:2em;
+}
+
```
Comment (Naman-Saxena1) [inline src/pages/home_page/home_page.css:27]: If possible prefer to avoid complex css selectors, otherwise it is fine.

**Review by Naman-Saxena1 — CHANGES_REQUESTED** [review]
Code looked pretty good to me, great work 😄 
I have suggested some things.
I also saw some commented code, if you are planning to use them in future then I guess its fine.
In NavBar, since "Artsy Shop" redirects to home page, you can add cursor: pointer to indicate to user that it is clickable.
Also, it will be easier for reviewer if you can list "Files to review" in review description.

File: src/backend/db/users.js [hunk 3]
```diff
@@ -6,7 +6,7 @@ import { formatDate } from "../utils/authUtils";
  * You can add default users of your wish with different attributes
  * Every user will have cart (Quantity of all Products in Cart is set to 1 by default), wishList by default
  * */
-
+/* eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI0MDdiMzQxYy01MWM5LTRlNjItYTVhYi0yODFlNGMwNzhhN2EiLCJlbWFpbCI6ImFkYXJzaGJhbGlrYUBnbWFpbC5jb20iLCJpYXQiOjE2NDczNDE0OTZ9.ur4ItGq2Fgjwh5ScrInACxAnCV8tcbgfVgT0JQF605Q */
```
Comment (iRohitGaur) [inline src/backend/db/users.js:9]: suggestion: remove token from code

File: src/context/ProductContext.jsx (hunk 1 above)
Comment (iRohitGaur) [inline src/context/ProductContext.jsx:22]: suggestion: remove commented code

File: src/context/ProductContext.jsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
+import axios from "axios";
+import {
+  createContext,
+  useContext,
+  useReducer,
+} from "react";
+import { useFetch } from "../custom_hooks/useAxios";
```
Comment (iRohitGaur) [inline src/context/ProductContext.jsx:7]: appreciation: nice use of a custom hook here

File: src/context/ProductContext.jsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  //   }
+  // };
+
+  const fetchCart = async () => {
+    try {
+      const res = await axios("/api/user/cart", {
+        method: "GET",
+        headers: {
+          authorization:
+            "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI4MmIxNjYwYi0yMTg2LTRiMmEtOTRlOC1mYzNmODUyYjcxNTQiLCJlbWFpbCI6ImpvaG5kb2VAZ21haWwuY29tIiwiaWF0IjoxNjQ3MzQ3OTY1fQ.CsaRbb8b9jWPRPUMY8OIt-TWE1bcRbFiUDz22YbNAoY",
```
Comment (iRohitGaur) [inline src/context/ProductContext.jsx:30]: suggestion: do not keep auth token in code

File: src/context/ProductContext.jsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+        return;
+      default:
+        return;
+    }
+  };
+  const [productState, productDispatch] = useReducer(
+    product_reducer_fn,
+    initialProductState
+  );
+  console.log("this is productstate", productState);
```
Comment (iRohitGaur) [inline src/context/ProductContext.jsx:64]: suggestion: remove `console.log`

File: src/context/ProductContext.jsx [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    <ProductContext.Provider value={{ productState, productDispatch }}>
+      {children}
+    </ProductContext.Provider>
+  );
+};
+
+const useProductContext = () => {
+  const context = useContext(ProductContext);
+  if (!context) {
+    throw new Error(`useProductContext must be used inside a context provider`);
```
Comment (iRohitGaur) [inline src/context/ProductContext.jsx:75]: appreciation: nice to see this error handling here

File: src/context/ProductsFilterContext.jsx [hunk 8]
```diff
@@ -0,0 +1,13 @@
+const ProductFilterContext = createContext()
+const filter_product_reducer = (state, { type, payload }) => {
+    switch (type) {
+        case "SEARCH":
+            return { ...state, }
+        case "FILTERBY":
+            return;
+        case "SortBY":
```
Comment (iRohitGaur) [inline src/context/ProductsFilterContext.jsx:8]: suggestion: to keep consistency, do all caps on SortBY as well

File: src/context/UserContext.jsx [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+      const res = await axios.post("/api/auth/login", {
+        email: "adarshbalika@gmail.com",
+        password: "adarshBalika123",
+      });
+      try {
+        console.log(res.data, "");
+      } catch {
+        console.log("error");
+      }
+    })(); */
```
Comment (iRohitGaur) [inline src/context/UserContext.jsx:35]: suggestion: remove commented code

File: src/pages/auth/utils/signupHandler.js [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+      const response = await axios.post(`/api/auth/signup`, {
+        firstName: "Adarsh",
+        lastName: "Balika",
+        email: "adarshbalika@neog.camp",
+        password: "adarshBalika",
+      });
+      // saving the encodedToken in the localStorage
+      localStorage.setItem("token", response.data.encodedToken);
+    } catch (error) {
+      console.log(error);
```
Comment (iRohitGaur) [inline src/pages/auth/utils/signupHandler.js:12]: suggestion: remove console.log

File: src/pages/cart_page/CartPage.jsx [hunk 11]
```diff
@@ -1,8 +1,12 @@
 import "./cart_page.css";
+import { useEffect } from "react";
 import { HorizontalCard } from "../../components";
 import { TotalBill } from "./component/TotalBill";
+import { useProductContext } from "../../context";
 
 export const CartPage = () => {
+  const { productState } = useProductContext();
+//   console.log(productState.cart);
```
Comment (iRohitGaur) [inline src/pages/cart_page/CartPage.jsx:9]: suggestion: remove commented code

File: src/pages/cart_page/component/TotalBill.jsx [hunk 12]
```diff
@@ -1,15 +1,17 @@
 import React from "react";
+import { useProductContext } from "../../../context";
 
 export const TotalBill = () => {
-  const cart = useCart();
+  const {productState} = useProductContext();
+  // let  total += productState.cart.map(item=>item.price);
```
Comment (iRohitGaur) [inline src/pages/cart_page/component/TotalBill.jsx:6]: suggestion: remove commented code

File: src/pages/home_page/components/CategoriesSection.jsx [hunk 13]
```diff
@@ -16,16 +16,16 @@ export const CategoriesSection = () => {
...
+  // const category_list = [
+  //   "Abstraction",
+  //   "Nature",
+  //   "Landscape",
+  //   "urban",
+  //   "Street",
+  //   "Medival",
+  //   "Pop Culture",
+  //   "Portrait",
+  // ];
```
Comment (iRohitGaur) [inline src/pages/home_page/components/CategoriesSection.jsx:28]: remove commented code

**Comment by iRohitGaur** [comment]
cta

suggestion: change your button color as it is not looking like a CTA until we hover it.

**Review by iRohitGaur — APPROVED** [review]
UI looks great. Nice work. Good to merge with changes 👍🏻

File: src/context/ProductContext.jsx (hunk 4 above)
Comment (bhtibrewal) [reply src/context/ProductContext.jsx:7]: thanks

File: src/context/ProductContext.jsx (hunk 5 above)
Comment (bhtibrewal) [reply src/context/ProductContext.jsx:30]: ok i will remove it

File: src/context/ProductContext.jsx (hunk 6 above)
Comment (bhtibrewal) [reply src/context/ProductContext.jsx:64]: just for testing, for now, will remove later

File: src/context/ProductContext.jsx (hunk 7 above)
Comment (bhtibrewal) [reply src/context/ProductContext.jsx:75]: thanks, learned from kent. c blogs

File: src/context/ProductsFilterContext.jsx (hunk 8 above)
Comment (bhtibrewal) [reply src/context/ProductsFilterContext.jsx:8]: corrected

File: src/context/UserContext.jsx (hunk 9 above)
Comment (bhtibrewal) [reply src/context/UserContext.jsx:35]: removed

File: src/pages/auth/utils/signupHandler.js (hunk 10 above)
Comment (bhtibrewal) [reply src/pages/auth/utils/signupHandler.js:12]: done

File: src/pages/cart_page/CartPage.jsx (hunk 11 above)
Comment (bhtibrewal) [reply src/pages/cart_page/CartPage.jsx:9]: done

File: src/pages/cart_page/component/TotalBill.jsx (hunk 12 above)
Comment (bhtibrewal) [reply src/pages/cart_page/component/TotalBill.jsx:6]: have to use it

File: src/context/ProductContext.jsx (hunk 1 above)
Comment (bhtibrewal) [reply src/context/ProductContext.jsx:22]: yes, I have removed the code that is not needed some I need

**Comment by bhtibrewal** [comment]
> Code looked pretty good to me, great work 😄 I have suggested some things. I also saw some commented code, if you are planning to use them in future then I guess its fine. In NavBar, since "Artsy Shop" redirects to home page, you can add cursor: pointer to indicate to user that it is clickable. Also, it will be easier for reviewer if you can list "Files to review" in review description.

Thank naman for the feedbacks

**Pull request merged**